[libpas] Move pas_zero_memory into its own file

Achierius opened the pull request:
#### bb082ecc001a8320b3f50c7d566db61204e082aa
<pre>
[libpas] Move pas_zero_memory into its own file
<a href="https://bugs.webkit.org/show_bug.cgi?id=299637">https://bugs.webkit.org/show_bug.cgi?id=299637</a>
<a href="https://rdar.apple.com/161439504">rdar://161439504</a>

Reviewed by Mark Lam and Yusuke Suzuki.

This is preparatory work for landing the MTE patch, as it breaks up a
dependency cycle that would otherwise be introduced between pas_mte.h
and pas_utils.h.

Canonical link: <a href="https://commits.webkit.org/300713@main">https://commits.webkit.org/300713@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ae2ef9f2318a1e01df56722790d02ab9897a9ef4

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/123583 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/43298 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/33994 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/130350 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/75726 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/52eb2cd6-0363-4799-80a7-bfe03a6bb4bb) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/44021 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/51892 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/93967 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/62371 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/81b322fa-bfc9-4bf7-bab6-de559c75426c) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/126536 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/35067 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/110560 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/74572 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/0b3668c3-60ff-46dd-8e79-7218a2a07475) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/34036 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/28718 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/73827 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/115742 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/104786 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/28942 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/133041 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/122115 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/50534 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/38474 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/102442 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/50909 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/106781 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/102285 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26000 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/47637 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/25875 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/47345 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/50388 "Built successfully") | | [❌ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/152469 "Hash ae2ef9f2 for PR 51411 does not build (failure)") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/49862 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/35/builds/152469 "Hash ae2ef9f2 for PR 51411 does not build (failure)") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/53209 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/51537 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

MenloDorian commented on Source/bmalloc/libpas/src/libpas/pas_zero_memory.h:
```diff
@@ -0,0 +1,37 @@
+
+#include "pas_utils.h"
+#include <stdint.h>
+
```
You probably need to add `PAS_BEGIN_EXTERN_C;` here and `PAS_END_EXTERN_C;` at the bottom.

Also a cursory check of other libpas header files shows that they rely on classic header guard e.g.
```
#ifndef PAS_HEAP_H
#define PAS_HEAP_H
...
#endif /* PAS_HEAP_H */
```

... instead of `#pragma once`.  Can you make this header file do the same for consistency?

MenloDorian reviewed (approved):
r=me with issues addressed and if EWS is green.

Achierius committed:
[libpas] Move pas_zero_memory into its own file
https://bugs.webkit.org/show_bug.cgi?id=299637
rdar://161439504

Reviewed by Mark Lam and Yusuke Suzuki.

This is preparatory work for landing the MTE patch, as it breaks up a
dependency cycle that would otherwise be introduced between pas_mte.h
and pas_utils.h.

Canonical link: https://commits.webkit.org/300713@main
